lint-scan.php

```diff
@@ -455,8 +360,295 @@ function vipgoci_lint_scan_commit(
+		$options,
+		$filename,
+		'linted'
+	);
+
+	/*
+	 * Actually lint the file
+	 */
+
+	vipgoci_log(
+		'About to PHP-lint file',
+
+		array(
+			'repo_owner'     => $repo_owner,
+			'repo_name'      => $repo_name,
+			'commit_id'      => $commit_id,
+			'filename'       => $filename,
+			'temp_file_name' => $temp_file_name,
+		)
+	);
+
+	$file_issues_arr_raw = vipgoci_lint_do_scan_file(
+		$options['php-path'],
+		$temp_file_name
+	);
+
+	/* Get rid of temporary file */
+	unlink( $temp_file_name );
+
+
+	/*
+	 * Process the results, get them in an array format
+	 */
+
+	$file_issues_arr = vipgoci_lint_parse_results(
+		$filename,
+		$temp_file_name,
+		$file_issues_arr_raw
+	);
+
+	vipgoci_log(
+		'Linting issues details',
+		array(
+			'repo_owner'          => $repo_owner,
+			'repo_name'           => $repo_name,
+			'commit_id'           => $commit_id,
+			'filename'            => $filename,
+			'temp_file_name'      => $temp_file_name,
+			'file_issues_arr'     => $file_issues_arr,
+			'file_issues_arr_raw' => $file_issues_arr_raw,
+		),
+		2
+	);
+
+	/* Get rid of the raw version of issues */
+	unset( $file_issues_arr_raw );
+
+
+	return $file_issues_arr;
+
+}
+
+
+/**
+ * This signature is not ready
+ *
+ * This function will be split into different functions that can be moved to
+ * other files (eg git-repo.php for git function calls)
+ *
+ * @param array $prs_implicated
+ * @param array $filter_extensions
+ * @param string $filter_skip_folders
+ * @param array $options
+ * @param array $commit_skipped_files
+ * @param array $commit_issues_stats
+ * @param array $commit_issues_submit
+ *
+ * @return array
+ */
+function vipgoci_gitrepo_scan_modified_files_only( array $prs_implicated, array $filter_extensions = [], string $filter_skip_folders = '', array $options, array &$commit_skipped_files, array &$commit_issues_stats, array &$commit_issues_submit  ): array {
+
+	/**
+	 * Get modified files in each PR that is implicated by the commit
+	 * Iterate it and compare base with head
+	 */
+
+
+	$local_git_repo = $options['local-git-repo'];
+	$prs_modified_files = array();
+
+	/**
+	 * @todo split param arrays and mout the desired strings:
+	 * -- '\'*.php\' '
+	 * -- '\':!tests1/*\' \':!tests/*\''
+	 */
+	$filter_extensions = '-- \'*.php\'';
+//	$filter_skip_folders = '-- \':!tests1/*\'';
+
+	/**
+	 * Save current branch so that we can checkout it at the end of the process
+	 */
+	$cmd     = sprintf(
+		"git -C %s branch --show-current 2>&1",
+		$local_git_repo
+	);
+	$local_git_branch = shell_exec($cmd);
+
+	foreach ($prs_implicated as $pr_number => $pr) {
+		/**
+		 * @todo these attributions are not required
+		 */
+		$pr_base_ref    = $pr->base->ref; // develop main
+		$pr_head_ref    = $pr->head->ref; // branches that contains that commit at the top of the tree, and has open PRs
+
+		/**
+		 * Checkout PR base branch so that git local gets information to compare
+		 */
+		$cmd     = sprintf(
+			"git -C %s checkout %s 2>&1",
```

Comment:
If we are to compare branches rather than commits, I don't see ways to bypass this with ``git``. 
The refs should be available locally in order to be used for git log/git diff, etc. 
Having said that, I'm not sure if GH API contains an endpoint that provides the information that it's required here, I don't think it does.

Reply:
The only way would be checking the files in the GH API running a GET request for each implicated PR:

E.g., https://api.github.com/repos/ariskataoka/vipgoci-tests/pulls/22/files

And then, filtering them manually.